contracts/Remittance.sol

```diff
@@ -1,16 +1,137 @@
 //SPDX-License-Identifier: MIT
 pragma solidity >= 0.6.0 < 0.7.0;
 
-import "../node_modules/@openzeppelin/contracts/access/Ownable.sol";
-import "../node_modules/@openzeppelin/contracts/utils/Pausable.sol";
+import "./access/Ownable.sol";
+import "./utils/Pausable.sol";
 
-import {SafeMath} from "../node_modules/@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeMath} from "./math/SafeMath.sol";
 
 contract Remittance is Ownable, Pausable {
 
-    using SafeMath for uint;
+    using SafeMath for uint256;
 
+    mapping(bytes32 => RemittanceInstance) public remittances;
 
+    // The 'valid' element provides a flag for checking a bonafide hash key
+    struct RemittanceInstance {
+        address funder;
+        address broker;
+        uint256 fundsOwed;
+        uint256 expiryDate;
+        bool valid;
+    }
+
+    event RemittanceCreated(
+        bytes32 indexed hash,
+        address indexed funder,
+        address indexed broker,
+        uint256 amount,
+        uint256 expiryDate
+    );
+
+    event RemittanceFundsReleased(
+        bytes32 indexed hash,
+        address indexed broker,
+        uint256 amount
+    );
+
+    event RemittanceFundsReclaimed(
+        bytes32 indexed hash,
+        address indexed funder,
+        uint256 amount
+    );
+
+    // The funder can create a remittance
+    // Hash is a sha256 hash of the broker address + recipient password
+    function create(bytes32 recipientPasswordHash, address broker, uint256 expiryDate)
+    external
+    payable
+    whenNotPaused
+    returns(bool success)
+    {
+        require(broker != msg.sender, "The caller cannot be the broker");
+        require(msg.value > 0, "The amount must be greater than 0");
+
+        bytes32 rehash = keccak256ReHash(recipientPasswordHash, broker);
+
+        RemittanceInstance storage remittanceInstance = remittances[rehash];
+
+        remittanceInstance.funder = msg.sender;
+        remittanceInstance.broker = broker;
+        remittanceInstance.fundsOwed = msg.value;
+        remittanceInstance.expiryDate = expiryDate;
+        remittanceInstance.valid = true;
+
+        emit RemittanceCreated(rehash, msg.sender, broker, msg.value, expiryDate);
+
+        return true;
+    }
+
+    // The broker sends the recipient's password to release the funds
+    function release(bytes32 recipientPasswordHash) external whenNotPaused returns(bool success) {
+
+        bytes32 rehash = keccak256ReHash(recipientPasswordHash, address(msg.sender));
+
+        //Retrieve remittance
+        RemittanceInstance storage remittanceInstance = remittances[rehash];
+
+        uint256 fundsOwed = remittanceInstance.fundsOwed;
+
+        require(remittanceInstance.valid, "No remittance found");
+        require(fundsOwed > 0, "No funds available");
+
+        require(block.timestamp < remittanceInstance.expiryDate, "The remittance has expired");
```

Comment:
You chose to limit the `release` also by the deadline. It is ok, although it is a matter of spec. You could imagine that the spec would enforce a deadline only on `reclaim`.

Reply:
Also, note that the `block.timestamp` is an approximative measure that is controlled by the miner. As opposed to `block.number`, which is a strict measure not controlled by the miner whose moment in time is approximate. 
Nothing's perfect.

Reply:
OK, Thank you. That is interesting. I have been reading on the difference in more detail. In this context, the block.timestamp appears to be sufficient as the absolute accuracy of the expiry date is not critical. It is within acceptable approximation.